File: engine/transpile/src/test/resources/golden/SYK007/HOST_商品コード.py
"""SYK007 のデータ部レコード HOST-商品コード を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class HOST_商品コード:
    """01 HOST-商品コード 総バイト長 8。"""

    LENGTH = 8

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(8)
        else:
            self._data = bytearray(data)
            if len(self._data) < 8:
                self._data.extend(b"\x00" * (8 - len(self._data)))

    @property
    def data(self):
        return self._data

    # HOST-商品コード
    def get_HOST_商品コード(self):
        return decode_alphanumeric(self._data, 0, 8)
    def set_HOST_商品コード(self, value):
        encode_alphanumeric(self._data, 0, 8, value)

File: engine/transpile/src/test/resources/golden/SYK007/HOST_商品コード.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK007 のデータ部レコード HOST-商品コード を逐語対訳した自動生成コード。 */
public final class HOST_商品コード {

    /** 01 HOST-商品コード 総バイト長 8。 */
    public static final int LENGTH = 8;

    private final byte[] data;

    public HOST_商品コード() {
        this.data = new byte[8];
    }

    public HOST_商品コード(byte[] source) {
        this.data = Arrays.copyOf(source, 8);
    }

    public byte[] data() {
        return data;
    }

    // HOST-商品コード
    public String get_HOST_商品コード() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 8);
    }
    public void set_HOST_商品コード(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 8, value);
    }
}
